pkg/lib/authn/user/model_test.go

```diff
@@ -87,5 +88,26 @@ func TestAccountStatus(t *testing.T) {
 			_, err = anonymized.Anonymize()
 			So(err, ShouldBeError, "invalid account status transition: anonymized -> anonymized")
 		})
+
+		Convey("scheduled anonymization by admin", func() {
+			scheduledAnonymization := AccountStatus{
+				IsDisabled:  true,
+				AnonymizeAt: &anonymizeAt,
+			}
+			var err error
+
+			_, err = scheduledAnonymization.Disable(nil)
+			So(err, ShouldBeError, "invalid account status transition: scheduled_anonymization_disabled -> disabled")
+
+			_, err = scheduledAnonymization.Reenable()
+			So(err, ShouldBeError, "invalid account status transition: scheduled_anonymization_disabled -> normal")
+
+			_, err = scheduledAnonymization.ScheduleAnonymizationByAdmin(deleteAt)
+			So(err, ShouldBeError, "invalid account status transition: scheduled_anonymization_disabled -> scheduled_anonymization_disabled")
+
+			normal, err := scheduledAnonymization.UnscheduleAnonymizationByAdmin()
+			So(err, ShouldBeNil)
+			So(normal.Type(), ShouldEqual, AccountStatusTypeNormal)
```

Comment:
Can we demonstrate all other possible state transitions? For example, demonstrate whether it is possible to have scheduled_anonymization_disabled -> scheduled_deletion_disabled, scheduled_deletion_disabled -> scheduled_anonymization_disabled, etc.